git_machete/git_operations.py

```diff
@@ -373,41 +377,335 @@ def load_all_reflogs(self) -> None:
             if len(branch_and_pos) != 2:  # invalid, shouldn't happen
                 continue
             b, pos = branch_and_pos
-            if b not in self.reflogs_cached:
-                self.reflogs_cached[b] = []
-            self.reflogs_cached[b] += [(sha, subject)]
+            if b not in self.__reflogs_cached:
+                self.__reflogs_cached[b] = []
+            self.__reflogs_cached[b] += [(sha, subject)]
 
     def reflog(self, b: str) -> List[REFLOG_ENTRY]:
         # git version 2.14.2 fixed a bug that caused fetching reflog of more than
         # one branch at the same time unreliable in certain cases
         if self.get_git_version() >= (2, 14, 2):
-            if self.reflogs_cached is None:
-                self.load_all_reflogs()
-            return self.reflogs_cached.get(b, [])
+            if self.__reflogs_cached is None:
+                self.__load_all_reflogs()
+            return self.__reflogs_cached.get(b, [])
         else:
-            if self.reflogs_cached is None:
-                self.reflogs_cached = {}
-            if b not in self.reflogs_cached:
+            if self.__reflogs_cached is None:
+                self.__reflogs_cached = {}
+            if b not in self.__reflogs_cached:
                 # %H - full hash
                 # %gs - reflog subject
-                self.reflogs_cached[b] = [
+                self.__reflogs_cached[b] = [
                     tuple(entry.split(":", 1)) for entry in utils.non_empty_lines(  # type: ignore
                         # The trailing '--' is necessary to avoid ambiguity in case there is a file called just exactly like the branch 'b'.
                         self.popen_git("reflog", "show", "--format=%H:%gs", b, "--"))
                 ]
-            return self.reflogs_cached[b]
+            return self.__reflogs_cached[b]
 
     def create_branch(self, b: str, out_of_revision: str) -> None:
         self.run_git("checkout", "-b", b, out_of_revision)
         self.flush_caches()  # the repository state has changed b/c of a successful branch creation, let's defensively flush all the caches
 
     def flush_caches(self) -> None:
         self.branch_defs_by_sha_in_reflog = None
```

Comment:
Huh this field is veeery Machete-specific... I suppose it's located in GitContext for the sake of easier cache flushing right?

Reply:
TBH I'd move this attr to `MacheteClient`, make private, and add a method `flush_caches` in `MacheteClient` that would clear `branch_defs_by_sha_in_reflog` and:
* either call `flush_caches()` on `GitContext` OR
* just discard the old `GitContext` and create a new one? that way `GitContext.flush_caches` method won't even be necessary

Reply:
yes, it was to make flushing easier, in one place. Because `flush_caches` is used later in `GitContext`, i'd like to call `flush_caches` on `GitContext` from `MacheteClient`.